src/main/java/com/azure/cosmos/kafka/connect/sink/ExceptionsHelper.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.kafka.connect.sink;
+
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.HttpConstants;
+
+public class ExceptionsHelper {
+    public static boolean canBeTransientFailure(int statusCode, int substatusCode) {
+        return statusCode == HttpConstants.StatusCodes.GONE
+                || statusCode == HttpConstants.StatusCodes.SERVICE_UNAVAILABLE
+                || statusCode == HttpConstants.StatusCodes.INTERNAL_SERVER_ERROR
+                || statusCode == HttpConstants.StatusCodes.REQUEST_TIMEOUT
+                || (statusCode == HttpConstants.StatusCodes.NOTFOUND && substatusCode == 1002);
```

Comment:
for 1002, we should have a constant mapping.

Reply:
GoneException 410 is missing - we still throw it from Java SDK in some cases

Reply:
Updated to use constant mapping for 1002
410 is added